python/pylibcudf/pylibcudf/tests/test_transform.py

```diff
@@ -81,3 +83,33 @@ def test_one_hot_encode():
         schema=pa.schema([pa.field("", pa.bool_(), nullable=False)] * 3),
     )
     assert result.equals(expected)
+
+
+def test_transform_udf():
+    @cuda.jit(device=True)
+    def op(a, b, c):
+        return (a + b) * c
```

Comment:
I think it'd be better to remove numba all together from this test. But if we can't, we should make it an optional test dependency _I think_.
CC @vyasr for his thoughts
```suggestion
def test_transform_udf():
    pytest.importorskip("numba")
    import numba
    from numba import cuda

    @cuda.jit(device=True)
    def op(a, b, c):
        return (a + b) * c
```

Reply:
I don't feel too strongly that we have to keep testing dependencies to a minimum as long as we don't impose requirements on the package itself. I feel like including PTX in Python when the generation is this straightforward feels like overkill to me.

Regarding optional dependencies in the test suite, I used to be a proponent of that but I think it winds up not being a bit error-prone because it's very easy to forget to install them at test time and then wind up accidentally skipping the test. If we want to use an optional testing dependency list we need to make sure that we are careful to have that be an additional extra in pyproject.toml that we install in CI. For conda we probably don't bother splitting the test dependency list in dependencies.yaml. I'm OK with that choice if we want to go that route, but we have to make sure the test isn't skipped anywhere in our CI.

Reply:
Yeah, pylibcudf already uses numba internally for some of it's UDFs

Reply:
> I don't feel too strongly that we have to keep testing dependencies to a minimum as long as we don't impose requirements on the package itself. 

Me neither. So we can leave the test alone. Definitely looks better without a PTX dump.
> 
> Regarding optional dependencies in the test suite, I used to be a proponent of that but I think it winds up not being a bit error-prone because it's very easy to forget to install them at test time and then wind up accidentally skipping the test. If we want to use an optional testing dependency list we need to make sure that we are careful to have that be an additional extra in pyproject.toml that we install in CI. For conda we probably don't bother splitting the test dependency list in dependencies.yaml. I'm OK with that choice if we want to go that route, but we have to make sure the test isn't skipped anywhere in our CI.

Ok thanks for explaining that. Do you know of an example of where we maintain an optional testing dependency list in cudf or other RAPIDS projects?

Reply:
Also this discussion is non-blocking for this PR.

Reply:
Not off the top of my head. I know there are some `pytest.importorskip` blocks in cuml or cugraph, but I don't know if those packages are left out of package metadata altogether or if we just always install them as part of the `test` extra in pyproject.toml and the tests dependency group from dependencies.yaml for conda deps.

Reply:
Can I resolve this conversation?